spmd/compiler/fusion.py

```diff
@@ -171,29 +170,23 @@ def _create_fusion_buffers(
     appends to GlobalInfo if passed in"""
 
     # default to inserting just after last placeholder node
+    # TODO - more efficient if we drop the buffer right before first use
+    # to reduce memory pressure.
     for node in gm.graph.nodes:
         if node.op == OP.PLACEHOLDER:
             continue
         insert_before_node = node
         break
 
-    # TODO - fix with correct rank - needs to match with higher DTensor device
-
-    rank = dist.get_rank()
-    if torch.distributed.is_initialized():
-        torch.cuda.set_device(rank)
-    rank_device = torch.cuda.current_device()
-
     ring_buffer = []
     new_buffer_node = None
     with gm.graph.inserting_before(insert_before_node):
         for i in range(ring_size):
             new_buffer_node = gm.graph.create_node(
                 OP.CALL_FUNCTION,
                 target=torch.empty,
-                # TODO - need device from DTensor to put buffer on gpu
                 args=(buffer_size,),
-                kwargs={"device": rank_device},
+                kwargs={"device": "cuda"},
```

Comment:
does this assume a set_device(rank) call? if so can we make a comment regarding that?

Reply:
have added a comment for this.  
From what I could see, DTensor uses the same exact concept of referencing "cuda", and per the docs here "cuda" is auto-expanded to "cuda:x" where X is the device id, meaning DTensor should fail earlier if this is not properly set and we can thus rely on it. 
 
"e.g., a [torch.Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor) constructed with device 'cuda' is equivalent to 'cuda:X' where X is the result of [torch.cuda.current_device()](https://pytorch.org/docs/stable/generated/torch.cuda.current_device.html#torch.cuda.current_device)."